src/librepgp/stream-key.cpp

```diff
@@ -721,6 +726,10 @@ process_pgp_key_auto(pgp_source_t &          src,
                            PGP_PKT_SECRET_SUBKEY})) {
         ret = RNP_ERROR_READ;
     }
+    /* change error code if we didn't process anything at all */
```

Comment:
Looks like a workaround. What is the purpose of this?

Reply:
@rrrooommmaaa This is actually to protect from the case where it is impossible to read anything from the stream, since this function could be used to 'read  till the end of stream, skipping bad format errors'.

Reply:
@ni4 don't we have `pgp_source_t::error` for this?

Reply:
@rrrooommmaaa There is no error on the source in this case, since nothing is actually read - parser peeks few bytes to detect packet type on source, and then returns an error.

Reply:
@ni4 so, is there actually an error? What is the line number where parser detects error? Maybe, it's better to set `pgp_source_t::error` there?

Reply:
@rrrooommmaaa There is not error on source, but on source contents, i.e. everything is fine with reading but we cannot continue to process the data since contents are not something we expected.